VultisigApp/VultisigApp/Views/FunctionCall/FunctionCallView.swift

```diff
@@ -131,7 +138,19 @@ struct FunctionCallView: View {
     
     private func setData() async {
         if let coin = coin {
-            tx.coin = coin
+            // Only update if the passed coin is different from current tx.coin
+            // This preserves existing behavior while fixing the BTC default issue
+            if tx.coin.id != coin.id {
+                tx.coin = coin
+            }
+        } else {
+            // FALLBACK: If no coin provided and current coin is BTC default,
+            // try to find a native token from the vault
+            if tx.coin.id == Coin.example.id {
```

Comment:
why use Coin.example here? we should not use Coin.example, it is created to help with preview UI rendering